daemon/checkpoint.go

```diff
@@ -28,16 +28,22 @@ func (daemon *Daemon) CheckpointCreate(name string, config types.CheckpointCreat
 	}
 
 	var checkpointDir string
+	var checkpointAbsDir string
 	if config.CheckpointDir != "" {
```

Comment:
Since this is repeated 3 times, might as well put it into a small routine.

Since there's also always a `os.Stat` following, What about something like this:
```
func getCheckpointDir(dir string, new bool) (base string, abs string, err error)
```

`new` is here to distinguish call from create and the other operations. if `new` is set then the routine should return an error if  `os.Stat` succeeds (whether the target is a directory or not), otherwise it should return an error is the target doesn't exist, is not a directory or any other system error.